osu.Game/Screens/Edit/EditorClock.cs

```diff
@@ -37,6 +39,23 @@ public EditorClock(ControlPointInfo controlPointInfo, double trackLength, Bindab
             TrackLength = trackLength;
         }
 
+        public override void ProcessFrame()
+        {
+            base.ProcessFrame();
+
+            if (CurrentTime >= TrackLength)
+            {
+                if (!PlaybackFinished)
+                {
+                    PlaybackFinished = true;
+                    Stop();
+                    Seek(TrackLength);
+                }
+            }
+            else
+                PlaybackFinished = false;
```

Comment:
This is super weirdly written. If the track is running then `false` gets assigned every frame, but when it finishes `true` gets assigned only once. It feels (a) way too complex (if it's not a bottleneck, and I'd be very surprised if it was, it should just assign every frame), and (b) if we're even doing microoptimisations like this then it should probably be the other way around since the `true` path is the (*very*) unlikely one.